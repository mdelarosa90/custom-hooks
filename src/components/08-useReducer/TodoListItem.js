import React from 'react'
import PropTypes from 'prop-types'
import './style.css';

const TodoListItem = ({ todo, index, handleDelete, handleToggle }) => {
    return (
        <>
            <li key={todo.id} className="list-group-item"><p onClick={handleToggle(todo.id)} className={`${todo.done && 'complete'}`}>{index + 1}. {todo.desc}</p><button className="btn btn-danger" onClick={handleDelete(todo.id)}>Borrar</button></li>
        </>
    )
}

TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired,
    index: PropTypes.number.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToggle: PropTypes.func.isRequired
}

export default TodoListItem;
