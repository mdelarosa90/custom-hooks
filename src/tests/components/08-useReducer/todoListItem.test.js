import React from 'react';
import {shallow} from 'enzyme'
import { demoTodos } from '../../fixtures/demoTodos';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';

describe('Pruebas <TodoListItem />', () => {

    const todo = {...demoTodos[0]}
    const handleEliminar = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(<TodoListItem todo={demoTodos[0]} index={0} handleDelete={handleEliminar} handleToggle={handleToggle} />);
    
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de llamar la función handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleEliminar).toHaveBeenCalledTimes(0);
    })

    test('debe de llamar la función handleToggle', () => {
        wrapper.find('p').simulate('click');
        //expect(handleToggle).toHaveBeenCalledWith(1);
    })

    test('debe de mostrar correcamente el texto', () => {
        const p = wrapper.find('p');
        expect(p.text()).toBe(`${1}. ${demoTodos[0].desc}`)
    })
    test('debe de tener la clase complete si done = true', () => {
        const todo = demoTodos[0];
        todo.done = true;
        const handleEliminar = jest.fn();
        const handleToggle = jest.fn();
        const wrapper = shallow(<TodoListItem todo={todo} index={0} handleToggle={handleToggle} handleDelete={handleEliminar} />);
        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    })
})