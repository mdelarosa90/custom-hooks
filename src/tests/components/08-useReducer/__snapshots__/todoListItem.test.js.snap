// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <TodoListItem /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1"
  >
    <p
      className="false"
    >
      1
      . 
      Aprender React
    </p>
    <button
      className="btn btn-danger"
    >
      Borrar
    </button>
  </li>
</Fragment>
`;
